draft-ietf-tls-rfc8446bis.md

```diff
@@ -3100,66 +3100,61 @@ in TLS 1.2 and below.
 In TLS 1.3, the server's SCT information is carried in an extension in the
 CertificateEntry.
 
-#### Server Certificate Selection
+#### Certificate Selection
 
-The following rules apply to the certificates sent by the server:
+The following rules apply to the certificates sent by the client or server:
 
 - The certificate type MUST be X.509v3 {{RFC5280}}, unless explicitly negotiated
   otherwise (e.g., {{RFC7250}}).
 
 - The end-entity certificate MUST allow the key to be used for signing with
-  a signature scheme indicated in the client's "signature_algorithms"
+  a signature scheme indicated in the peer's "signature_algorithms"
   extension (see {{signature-algorithms}}). That is, the digitalSignature bit
   MUST be set if the Key Usage extension is present, and the public key (with
   associated restrictions) MUST be compatible with some supported signature
   scheme.
 
-- The "server_name" {{RFC6066}} and "certificate_authorities" extensions are used to
-  guide certificate selection. As servers MAY require the presence of the "server_name"
-  extension, clients SHOULD send this extension when the server is identified by name.
+- If the peer sent a "certificate_authorities" extension, at least one of the
+  certificates in the certificate chain SHOULD be issued by one of the listed
+  CAs.
 
-All certificates provided by the server MUST be signed by a
-signature algorithm advertised by the client, if it is able to provide such
+The following rules additionally apply to certificates sent by the client:
+
+- If the CertificateRequest message contained a non-empty "oid_filters"
+  extension, the end-entity certificate MUST match the extension OIDs
+  that are recognized by the client, as described in {{oid-filters}}.
+
+The following rules additionally apply to certificates sent by the server:
+
+- The "server_name" {{RFC6066}} extension is used to guide certificate
+  selection. As servers MAY require the presence of the "server_name" extension,
+  clients SHOULD send this extension when the server is identified by name.
+
+All certificates provided by the sender MUST be signed by a
+signature algorithm advertised by the peer, if it is able to provide such
 a chain (see {{signature-algorithms}}).
 Certificates that are self-signed
 or certificates that are expected to be trust anchors are not validated as
 part of the chain and therefore MAY be signed with any algorithm.
 
 If the server cannot produce a certificate chain that is signed only via the
```

Comment:
"Server" or "sender"?
If the **server** cannot produce... then it SHOULD continue the handshake... not known to be supported by the **peer**.
Perhaps it should be either "server and client" or "sender and receiver" or "peer and peer".

Reply:
Ah yeah, this part is weirdly written. It's because of the client can decline to send a cert but the server can't. Rephrased to hopefully be clearer.